packages/discussions/graphql/resolvers/Discussion/comments.js

```diff
@@ -245,6 +244,29 @@ module.exports = async (discussion, args, context, info) => {
     }
   }
 
+  /* 
+    AUTO = comments are sorted 
+    - by date if the first comment was created in the last 24 hours 
+    - or by votes otherwise 
+    the property resolvedOrderBy is just needed when DiscussionOrder === AUTO
+  */
+  let resolvedOrderBy
+  if (orderBy === 'AUTO') {
+    const sortedComments = [...comments].sort(
```

Comment:
I was thinking, maybe a more intuitive way to get to the "oldest" entry – instead of sorting – could be a reducer. Would you feel comfortable with that, or is it contrary to my believe rather unintuitive?

Here is a mock:

```js
// Keep this rather a the top of file.
const OLD_THRESHOLD_IN_MS = 1000 * 60 * 60 * 24 // 24 hours

// Then, inside if(orderBy === 'AUTO') {...

const initialComment = comments?.reduce((oldestCommentYet, currentComment) => {
  // If there was no oldestComment, set current comment as oldest
  if (!oldestCommentYet) {
    return currentComment 
  }

  // If currentComment is older than oldestCommentYet, set current as
  // oldest. Otherwise keep oldestComment.
  return currentComment.createdAt < oldestCommentYet.endDate
    ? currentComment
    : oldestCommentYet
}, false) // Start with false. If comment array is empty, false will prevail

// 
// So, up until here, either intialComment is falsy or we got the oldest comment at hand.
// 

const oldThresholdInMs = new Date().getTime() - OLD_THRESHOLD_IN_MS

// If initialComment older than oldThresholdInMs, order by VOTES.
// And if initialComment is falsy or younger, order by DATE.
const resolvedOrderBy = (initialComment?.createdAt?.getTime() < oldThresholdInMs && 'VOTES') || 'DATE'
```

Reply:
Yes, I agree, much more elegant and also better in terms of performance I think